SIPS/sip-3.md

```diff
@@ -0,0 +1,165 @@
+---
+sip: 3
+title: Transaction Insights
+status: Review
+discussions-to: https://github.com/MetaMask/SIPs/discussions/31
+author: Hassan Malik (@hmalik88)
+created: 2022-08-23
+---
+
+## Table of Contents
+
+- [Table of Contents](#table-of-contents)
+- [Abstract](#abstract)
+- [Motivation](#motivation)
+- [Specification](#specification)
+  - [Language](#language)
+  - [Definitions](#definitions)
+  - [Common types](#common-types)
+  - [Snap Developer](#snap-developer)
+    - [Manifest](#manifest)
+    - [Snap](#snap)
+- [History](#history)
+- [Copyright](#copyright)
+
+
+## Abstract
+
+This SIP proposes a way for snaps to provide transaction insights that can then be displayed in the MetaMask confirmation UI.
+
+Example use-cases for transaction insights can be phishing detection, malicious address detection, transaction simulation, arbitrage trading.
+
+## Motivation
+
+One of the main use cases for transaction insights is transaction simulation. In the current state of the MetaMask wallet, we do not provide any sort of insight as to whether or not a transaction will succeed based on the web of contract/address interaction that can happen. Transaction insight snaps would provide the ability to calculate some sort of insights and return them to the wallet..
```

Comment:
```suggestion
Deciding what information to show before a user is prompted to sign a transaction and furthermore deciding which information is critical is a difficult problem. It may even be user-dependant. Similarly to how SIP-2 allows snaps to expand the keyrings that MetaMask support, this SIP aims to expand the options the user is given in which information they see before signing a transaction.

The current MetaMask extension already has a "transaction insights" feature that does transaction simulation and shows the result to the user. This SIP aims to expand on this feature, allowing the community to build snaps that provide **any form of "insight"** into a transaction. This insight will be shown in the MetaMask UI alongside any official insight provided by MetaMask itself.
```

Rewrote this a bit. Let me know what you think!

Reply:
Looks good!